### DISCUSSION TITLE
Running using systemd and stdout

### DISCUSSION THREAD

**Discussion by RandyLBryant** [Dan-in-CA/SIP#298] [Ideas]
I just setup the systemd run as documented.  There is a fair amount of output that gets printed to the system daemon.log (sip stdout).  I am trying to find those output statements and see if there is a way to turn them off except for error situations.  Where do I find the prints and is there already a recommended way to turn most of them off?  

Example output:

Dec 26 17:21:01 SprinklerSystem sip[391]: ::ffff:192.168.1.16:57987 - - [26/Dec/2022 17:21:01] "HTTP/1.1 GET /api/status" - 200 OK
Dec 26 17:21:02 SprinklerSystem sip[391]: ::ffff:192.168.1.16:57987 - - [26/Dec/2022 17:21:02] "HTTP/1.1 GET /api/log" - 200 OK
Dec 26 17:21:32 SprinklerSystem sip[391]: ::ffff:192.168.1.16:57990 - - [26/Dec/2022 17:21:32] "HTTP/1.1 GET /api/status" - 200 OK
Dec 26 17:21:32 SprinklerSystem sip[391]: ::ffff:192.168.1.16:57990 - - [26/Dec/2022 17:21:32] "HTTP/1.1 GET /api/log" - 200 OK

**Comment by Dan-in-CA** [comment]
Hi Randy,
They are responses to GET requests from the UI.
The code that generates them is in webpages.py lines 641 and 719
You could comment out the  return lines 716 & 748 but I think the UI would not show SIP activity correctly.

Dan

**Reply by RandyLBryant** [reply]
I wold still like to understand this but I found a better way I think.  I just use StandardOutut=file:/dev/null in sip3_service, to eliminate stdout in the daemon.log.  

For some reason the line numbers don't agree with my source file for webpages.py. 

My line number 641:

                   if not gv.sd["iw"][b] & 1 << s:
                        duration = duration * gv.sd["wl"] // 100 * extra_adjustment
                    gv.rs[sid][2] = duration

Line 716:

                            if sbit:
                                status["status"] = "on"
                                status["reason"] = "program"
                                status["programName"] = pname
                                status["remaining"] = rem
                            else:

**Reply by Dan-in-CA** [reply]
Sorry, I forgot that I based the line numbers on a development rev I am working on.
I think your solution is a good way to go. The API calls are needed by the UI to be able to display current status but keeping log files small will help prolong the life of the SD card.

**Reply by RandyLBryant** [reply]
That is why I was looking into it. (SD card).  Can you show me where those lines get printed to stdout?

**Reply by Dan-in-CA** [reply]
I am not aware of the source of that output. I didn't design the UI. It was contributed by a user a few years ago.
The output does not seem to be from a print statement in the SIP core program code. It may be from the web.py framework SIP is built on.

**Reply by Dan-in-CA** [reply]
Actually It looks like it might be coming from the cheroot web server SIP uses.
